be/src/util/brpc_stub_cache.cpp

```diff
@@ -169,7 +234,50 @@ StatusOr<std::shared_ptr<starrocks::LakeService_RecoverableStub>> LakeServiceBrp
         return Status::RuntimeError("init brpc http channel error on " + host + ":" + std::to_string(port));
     }
     _stub_map.insert(endpoint, stub);
+    _last_access_time_map.insert(endpoint, now);
     return stub;
 }
 
-} // namespace starrocks
\ No newline at end of file
+void LakeServiceBrpcStubCache::check_and_cleanup_expired_stubs(int32_t timeout_ms) {
+    std::lock_guard<SpinLock> l(_lock);
+
+    std::vector<butil::EndPoint> expired_endpoints;
+    auto now = std::chrono::steady_clock::now();
+    for (auto& entry : _last_access_time_map) {
+        auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(now - entry.second);
+        if (duration.count() >= timeout_ms) {
+            expired_endpoints.push_back(entry.first);
+        }
+    }
+
+    for (const auto& endpoint : expired_endpoints) {
+        LOG(INFO) << "cleanup lake service stub from endpoint:" << endpoint;
+        _stub_map.erase(endpoint);
+        _last_access_time_map.erase(endpoint);
+    }
+}
+
+BrpcStubManager::BrpcStubManager(ExecEnv* exec_env) : _exec_env(exec_env) {
+    _cleanup_thread = std::thread([this] {
+#ifdef GOOGLE_PROFILER
+        ProfilerRegisterThread();
+#endif
+        while (!_is_stopped.load()) {
+            LOG(INFO) << "Start to clean up expired brpc stubs";
+            _exec_env->brpc_stub_cache()->check_and_cleanup_expired_stubs(config::brpc_stub_expire_ms);
```

Comment:
better capture the configvar into a local var before doing the actual cleanup, to avoid inconsistent expiration in the same round.